app/src/main/java/com/github/dfa/diaspora_android/util/Helpers.java

```diff
@@ -84,6 +86,16 @@ public static File createImageFile() throws IOException {
         );
     }
 
+    public static Locale getLocaleByAndroidCode(String code) {
+        if (!TextUtils.isEmpty(code)) {
+            return code.contains("-r")
+                    ? new Locale(code.substring(0, 1), code.substring(4, 5)) // de-rAT
```

Comment:
Are you sure, this is right? Don't you have to take bigger substrings? (0,2),(4,6)